features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/trace/OtelTracerBuilderProviderTest.kt

```diff
@@ -446,4 +421,131 @@ internal class OtelTracerBuilderProviderTest {
     }
 
     // endregion
+
+    // region Sampling priority
+
+    @Test
+    fun `M not add a sample rate by default W creating a tracer`() {
+        // Given
+        val tracer = testedOtelTracerProviderBuilder.build()
+            .tracerBuilder(fakeInstrumentationName).build()
+
+        // When
+        val coreTracer: CoreTracer = tracer.getFieldValue("tracer")
+
+        // Then
+        val config: Config = coreTracer.getFieldValue("initialConfig")
+        val traceSampleRate: Double? = config.traceSampleRate
+        assertThat(traceSampleRate).isNull()
+    }
+
+    @Test
+    fun `M use the sample rate W setSampleRate`(@DoubleForgery(min = 0.0, max = 100.0) sampleRate: Double) {
+        // Given
+        val expectedNormalizedSampleRate = sampleRate / 100.0
+        val tracer = testedOtelTracerProviderBuilder.setSampleRate(sampleRate).build()
+            .tracerBuilder(fakeInstrumentationName).build()
+
+        // When
+        val coreTracer: CoreTracer = tracer.getFieldValue("tracer")
+
+        // Then
+        val config: Config = coreTracer.getFieldValue("initialConfig")
+        assertThat(config.traceSampleRate).isCloseTo(expectedNormalizedSampleRate, offset(0.005))
+    }
+
+    @Test
+    fun `M use user-keep priority W buildSpan { provided keep sample rate }`() {
+        // Given
+        val tracer = testedOtelTracerProviderBuilder
+            .setPartialFlushThreshold(1)
+            .setSampleRate(100.0)
+            .build()
+            .tracerBuilder(fakeInstrumentationName)
+            .build()
+
+        // When
+        val span = tracer
+            .spanBuilder(fakeOperationName)
+            .startSpan()
+        val delegateSpan: DDSpan = span.getFieldValue("delegate")
+        delegateSpan.forceSamplingDecision()
+        span.end()
+
+        // Then
+        val priority = delegateSpan.samplingPriority
+        assertThat(priority).isEqualTo(PrioritySampling.USER_KEEP.toInt())
+    }
+
+    @Test
+    fun `M use user-not-keep priority W buildSpan { provide not keep sample rate }`() {
+        // Given
+        val tracer = testedOtelTracerProviderBuilder
+            .setPartialFlushThreshold(1)
+            .setSampleRate(0.0)
+            .build()
+            .tracerBuilder(fakeInstrumentationName)
+            .build()
+
+        // When
+        val span = tracer
+            .spanBuilder(fakeOperationName)
+            .startSpan()
+        val delegateSpan: DDSpan = span.getFieldValue("delegate")
+        delegateSpan.forceSamplingDecision()
+        span.end()
+
+        // Then
+        val priority = delegateSpan.samplingPriority
+        assertThat(priority).isEqualTo(PrioritySampling.USER_KEEP.toInt())
```

Comment:
I believe it should be `PrioritySampling.USER_DROP` here by looking on the name of the test